com.unity.shadergraph/Editor/Data/Nodes/Input/Texture/ProceduralVirtualTextureNode.cs

```diff
@@ -21,8 +27,8 @@ public ProceduralVirtualTextureNode()
 
             vtProperty.displayName = "ProceduralVirtualTexture";
             vtProperty.overrideReferenceName = "MyPVT";
-            vtProperty.generatePropertyBlock = false;
             vtProperty.value.procedural = true;
+            vtProperty.value.bindPerMaterial = true;
```

Comment:
I agree that new nodes should indeed have bindPerMaterial=true as that is by far the most useful.
For existing shaders, I guess keeping it as-is is fine (though imho if we clearly document this, changing it is also a valid option).

Reply:
You can add a new node version, then on upgrade of old nodes, enable overrideHLSLDeclaration and set hlslDeclarationOverride to HLSLDeclaration.Global on the vtProperty.value.

Can look at Vector1ShaderProperty.cs as an example of versioning (though here you would version the PVT node):

        public override int latestVersion => 1;
        public override void OnAfterDeserialize(string json)
        {
            if (sgVersion == 0)
            {
                // change declaration values here
                ChangeVersion(1);
            }
        }

You will probably need to use OnAfterMultiDeserialize() instead of OnAfterDeserialize() to ensure that the vtProperty has been deserialized before you go changing it -- MultiDeserialize gets called after everything has been deserialized.